src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+        roleAuthorizer.grantRoleFromModule(uint8(Roles.BountyAdmin), addr);
+    }
+
+    /// @inheritdoc IBountyManager
+    function grantClaimAdminRole(address addr) external onlyAuthorized {
+        IRoleAuthorizer roleAuthorizer =
+            IRoleAuthorizer(address(__Module_proposal.authorizer()));
+        roleAuthorizer.grantRoleFromModule(uint8(Roles.ClaimAdmin), addr);
+    }
+
+    /// @inheritdoc IBountyManager
+    function grantVerifyAdminRole(address addr) external onlyAuthorized {
+        IRoleAuthorizer roleAuthorizer =
+            IRoleAuthorizer(address(__Module_proposal.authorizer()));
+        roleAuthorizer.grantRoleFromModule(uint8(Roles.VerifyAdmin), addr);
+    }
+
+    /// @inheritdoc IBountyManager
+    function revokeBountyAdminRole(address addr) external onlyAuthorized {
+        IRoleAuthorizer roleAuthorizer =
+            IRoleAuthorizer(address(__Module_proposal.authorizer()));
+        roleAuthorizer.revokeRoleFromModule(uint8(Roles.BountyAdmin), addr);
+    }
+
+    /// @inheritdoc IBountyManager
+    function revokeClaimAdminRole(address addr) external onlyAuthorized {
+        IRoleAuthorizer roleAuthorizer =
+            IRoleAuthorizer(address(__Module_proposal.authorizer()));
+        roleAuthorizer.revokeRoleFromModule(uint8(Roles.ClaimAdmin), addr);
+    }
+
+    /// @inheritdoc IBountyManager
+    function revokeVerifyAdminRole(address addr) external onlyAuthorized {
+        IRoleAuthorizer roleAuthorizer =
+            IRoleAuthorizer(address(__Module_proposal.authorizer()));
+        roleAuthorizer.revokeRoleFromModule(uint8(Roles.VerifyAdmin), addr);
+    }
+
+    //--------------------------------------------------------------------------
+    // {PaymentClient} Function Implementations
+
+    function _ensureTokenBalance(uint amount)
+        internal
+        override(PaymentClient)
+    {
+        uint balance = __Module_proposal.token().balanceOf(address(this));
+
+        if (balance < amount) {
+            // Trigger callback from proposal to transfer tokens
+            // to address(this).
+            bool ok;
+            (ok, /*returnData*/ ) = __Module_proposal.executeTxFromModule(
+                address(__Module_proposal.fundingManager()),
+                abi.encodeWithSignature(
+                    "transferProposalToken(address,uint256)",
+                    address(this),
+                    amount - balance
+                )
+            );
+
+            if (!ok) {
+                revert Module__PaymentClient__TokenTransferFailed();
+            }
+        }
+    }
+
+    function _ensureTokenAllowance(IPaymentProcessor spender, uint amount)
+        internal
+        override(PaymentClient)
+    {
+        IERC20 token = __Module_proposal.token();
+        uint allowance = token.allowance(address(this), address(spender));
+
+        if (allowance < amount) {
+            token.safeIncreaseAllowance(address(spender), amount - allowance);
```

Comment:
we gotta fork-test with multiple market leading ERC20 tokens to make sure that `safeIncreaseAllowance` works with them. As originally only `approve` was part of the ERC20 standard and not `safeIncreaseAllowance`.

Reply:
This should be tackled in #261 and #208